go/pserver/service.go

```diff
@@ -49,13 +53,58 @@ type Service struct {
 	optMap map[string]*optimizer
 }
 
+// Checkpoint saves the checkpoint for pserver
+type Checkpoint struct {
+	UUID      string `json:"uuid"`
+	MD5       string `json:"md5"`
+	Timestamp string `json:"timestamp"`
+	State     []byte
+	ParameterWithConfig
```

Comment:
agreed with @Yancey1989 . Do we need to separate `meta` and `data` in checkpoint ?